rust/connlib/tunnel/src/dns.rs

```diff
@@ -329,17 +382,23 @@ fn to_aaaa_records(ips: impl Iterator<Item = IpAddr>) -> Vec<AllRecordData<Vec<u
 
 fn build_dns_with_answer(
     message: Message<&[u8]>,
-    qname: DomainName,
     records: Vec<AllRecordData<Vec<u8>, DomainName>>,
 ) -> Result<Message<Vec<u8>>> {
+    // Take the original qname out of the message.
+    // DNS queries should always respond for the exact same qname that was queried, even if we expanded a single-label domain.
+    let qname = message
```

Comment:
This is interesting to point out. Previously, we took the parsed domain name from the query in passed it in separately to this function (despite the `message` being the query again).

Now that we turn single-label domains into FQDN, this actually broke the tests because upon querying for `app`, `connlib` suddenly returned `app.example.com` in the records.

I believe that is wrong and we should only ever return the exact domain name that was queried for, even if we internally expanded it to a FQDN.